### PR TITLE
Pomodoro function completed

### PR DESCRIPTION
**Pull request by Anik-Bardhan**
@mogiiee looked into issue #5 
Created a function which when called on discord will implement the Pomodoro method which is continuous concentration for 25 minutes and then a 5 minute break.
- [x] Set a timer for 25 minutes
- [x] Notify the user after 25 minutes
- [x] Set the timer for 5 minutes (starting of the break)
- [x] Notify the user after 5 minutes that Pomodoro has ended

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -84,11 +84,34 @@ async def daily_reminder(ctx, time):
...
+    message = await ctx.send("Your countdown has been started for 25 minutes")
+    while True:
+        mins, sec = divmod(time, 60)
+        time -= 1
+        if time == 0:
+            await message.edit(content=("Time remaining {:02d}:{:02d}".format(0, 0)))
+            await ctx.send(ctx.message.author.mention + " your countdown has ended!\nBreak starts.")
+            break
+
+        await message.edit(content=("Time remaining {:02d}:{:02d}".format(mins, sec)))
```
Comment (hg242322) [inline bot.py:98]: Please remove second-wise updates as it would result in too many requests to the server, which it might not be able to handle.

**Review by hg242322 — COMMENTED** [review]
Great work !!
Only a small change required.

File: bot.py (hunk 1 above)
Comment (Anik-Bardhan) [reply bot.py:98]: Okay on it. I actually added it by referring to the start_timer function in bot.py file as it has second-wise update.

**Comment by Anik-Bardhan** [comment]
@hg242322 did the changes for second-wise updates.

File: bot.py [hunk 2]
```diff
@@ -84,11 +84,30 @@ async def daily_reminder(ctx, time):
 @client.command(aliases=["start-pomodoro"])
 async def pomodoro(ctx, time):
     # Set a timer for 25 minutes
+    time = 1500
     # Notify the user after 25 minutes
+    message = await ctx.send("Your countdown has been started for 25 minutes")
+    while True:
+        time -= 1
+        if time == 0:
+            await message.edit(content=("Time remaining {:02d}:{:02d}".format(0, 0)))
```
Comment (hg242322) [inline bot.py:93]: Time remaining message is not required, as it would just appear and disappear instantly.

File: bot.py [hunk 3]
```diff
@@ -84,11 +84,30 @@ async def daily_reminder(ctx, time):
...
+        await asyncio.sleep(1)
     # Set the timer for 5 minutes (starting of the break)
-    # Notify the user after 5 minutes that Pomodoro has ended
-    pass
+    time = 300
+    await message.edit(content=("Your break countdown has been started for 5 minutes"))
+    while True:
+        time -= 1
+        if time == 0:
+            await message.edit(content=("Time remaining {:02d}:{:02d}".format(0, 0)))
```
Comment (hg242322) [inline bot.py:104]: Time Remaining message is not required.

File: bot.py [hunk 4]
```diff
@@ -84,11 +84,30 @@ async def daily_reminder(ctx, time):
...
 async def pomodoro(ctx, time):
     # Set a timer for 25 minutes
+    time = 1500
     # Notify the user after 25 minutes
+    message = await ctx.send("Your countdown has been started for 25 minutes")
+    while True:
+        time -= 1
+        if time == 0:
+            await message.edit(content=("Time remaining {:02d}:{:02d}".format(0, 0)))
+            await ctx.send(ctx.message.author.mention + " your countdown has ended!\nBreak starts.")
```
Comment (hg242322) [inline bot.py:94]: You may rephrase it as just one sentence, for example, Your Break has started.

File: bot.py [hunk 5]
```diff
@@ -84,11 +84,30 @@ async def daily_reminder(ctx, time):
...
     # Set the timer for 5 minutes (starting of the break)
-    # Notify the user after 5 minutes that Pomodoro has ended
-    pass
+    time = 300
+    await message.edit(content=("Your break countdown has been started for 5 minutes"))
+    while True:
+        time -= 1
+        if time == 0:
+            await message.edit(content=("Time remaining {:02d}:{:02d}".format(0, 0)))
+            await ctx.send(ctx.message.author.mention + " your break countdown has ended!")
```
Comment (hg242322) [inline bot.py:105]: Your Break has ended is repetitive, because we notify the user that their Pomodoro has ended.

File: bot.py [hunk 6]
```diff
@@ -84,11 +84,30 @@ async def daily_reminder(ctx, time):
 @client.command(aliases=["start-pomodoro"])
 async def pomodoro(ctx, time):
```
Comment (hg242322) [inline bot.py:85]: You may remove the time argument, since it is used nowhere.

**Review by hg242322 — CHANGES_REQUESTED** [review]
Just small refactoring is required.
You have done a Great Job and your PR is almost ready to be merged.

**Comment by Anik-Bardhan** [comment]
@hg242322 thank you. I have made the changes.

**Comment by hg242322** [comment]
Hey congratulations on getting the first PR merged, we would like to give you a shoutout on our twitter account twitter.com/ieeevitvellore. Can you please provide us with your twitter handle?

**Comment by Anik-Bardhan** [comment]
@hg242322 thank you for the merge. My twitter handle is [@anik_0406](https://twitter.com/anik_0406). Thank you once again.

**Pull request merged**